### DISCUSSION TITLE
Numbering the rows of a table

### DISCUSSION THREAD

**Discussion by ashkankiani** [komarovalexander/ka-table#236] [Q&A]
How can I have a numbered column of the number of rows? (Continuous counting 1,2,3,4....)
example: 
id Fname Lname
1     A          C
2     B          B
3     C         D

**Comment by komarovalexander** [comment]
You can add it to separate column. 
but if you dont want to modify your data there is an example with command column which shows columns not binded to data https://komarovalexander.github.io/ka-table/#/alert-cell you can show there whatever you want

**Reply by ashkankiani** [reply]
If we don't have an ID, it will be difficult to count. Your second method is the right solution, but how can you navigate the rows? Or the number of data objects. The solution is as follows.

`    const tablePropsInit = {
        columns: [
            {key: 'id', title: 'ID', isSortable: false, dataType: DataType.Number},
        }
        number: 1,
    };
`

`        cellText: {
            content: (props) => {
                switch (props.column.key) {
                    case 'id':
                        return tablePropsInit.number++
                }
            }
        },`

As you said, the data is not connected to the sets. Is there a solution for the first method?

**Reply by komarovalexander** [reply]
Do you need something like this?
https://stackblitz.com/edit/table-overview-ts-bcqbfb?file=Demo.tsx

**Reply by ashkankiani** [reply]
Yes, exactly. But I don't get the id from the object. I cannot consider ID as a key.
